app/tests/spec/views/settings.js

```diff
@@ -329,40 +541,88 @@ function (chai, $, sinon, View, RouterMock, TestHelpers,
             });
         });
       });
-    });
 
-    describe('communication preferences link', function () {
-      it('is visible if enabled', function () {
-        sinon.stub(view, 'isUserAuthorized', function () {
-          return p(true);
+      describe('hide success', function () {
+        it('displaySuccessUnsafe', function (done) {
+          view.SUCCESS_MESSAGE_DELAY = 5;
+          var spy = sinon.spy(view, 'hideSuccess');
+          view.render()
+            .then(function () {
+              view.displaySuccessUnsafe('hi');
+              setTimeout(function () {
```

Comment:
You might be able to do this a bit more gracefully using [p.delay](:

```js
return p().delay(10)
  .then(function () {
    assert.isTrue(spy.called, 'hideSuccessCalled');
  });
```
The promise will not be resolved until after the delay has elapsed, no need to use a try/catch, nor a `done`.